nixos/modules/config/networking.nix

```diff
@@ -223,7 +223,9 @@ in
             '';
 
       } // optionalAttrs config.services.resolved.enable {
-        "resolv.conf".source = "/run/systemd/resolve/resolv.conf";
+        # symlink the static version of resolv.conf as recommended by upstream:
+        # https://www.freedesktop.org/software/systemd/man/systemd-resolved.html#/etc/resolv.conf
+        "resolv.conf".source = "${pkgs.systemd}/lib/systemd/resolv.conf";
```

Comment:
An option for *not* symlinking `/etc/resolv.conf` at all could be added, but I don't see an easy way to do this, it would need a list of programs that can manage `resolv.conf` or so, so that `resolved`'s  symlink creation can be skipped…